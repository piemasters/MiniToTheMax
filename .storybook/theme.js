import { create } from "@storybook/theming/create";
import logo from "../content/assets/images/icon.png";


const theme = create({
    base: "light",

    colorPrimary: "#ff5470",
    colorSecondary: "#00dcb5",

    // UI
    appBg: "white",
    appContentBg: "white",
    appBorderColor: "grey",
    appBorderRadius: 10,

    // Typography
    fontBase: '"Open Sans", sans-serif',
    fontCode: "monospace",

    // Text colors
    textColor: "black",
    textInverseColor: "rgba(255,255,255,0.9)",

    // Toolbar default and active colors
    barTextColor: "white",
    barSelectedColor: "#00dcb5",
    barBg: "#00153d",

    // Form colors
    inputBg: "white",
    inputBorder: "white",
    inputTextColor: "black",
    inputBorderRadius: 4,

    brandTitle: "MiniToTheMax Storybook",
    brandUrl: "https://minitothemax.app",
    brandImage: logo
});

export default theme;