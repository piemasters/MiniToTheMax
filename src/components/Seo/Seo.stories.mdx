import {
  Story,
  Preview,
  Props,
  Source,
  Description,
  Meta,
} from '@storybook/addon-docs/blocks';
import { Seo } from './Seo';
import cover from './example-cover.jpg';

<Meta
  title="Components/Seo"
  component={Seo}
  parameters={{ jest: ['seo'] }}
  argTypes={{
    title: {
      name: 'title',
      description: 'Title of the page',
      defaultValue: 'SEO Title',
      control: { type: 'text' },
    },
    description: {
      name: 'description',
      description: 'A brief summary of the page',
      defaultValue: 'SEO description',
      control: { type: 'text' },
    },
    image: {
      name: 'image',
      description: 'Thumbnail used when link is shared',
      defaultValue: cover,
      control: { type: 'text' },
    },
    article: {
      name: 'article',
      description: 'Whether the page type is a blog post',
      defaultValue: false,
      control: { type: 'boolean' },
    },
    pathname: {
      name: 'pathname',
      description: 'The unique slug of the page after the base URL',
      defaultValue: '',
      control: { type: 'text' },
    },
  }}
/>

# Seo

With `MDX` we can define a story for `Seo` right in the middle of our
markdown documentation.

<Preview>
  <Story
    name="default"
    args={{
      title: 'SEO Title',
      description: 'Description of the SEO',
      image: cover,
      article: true,
      pathname: 'blog-post-1',
    }}
  >
    {(args) => <Seo {...args} />}
  </Story>
</Preview>

# Props

<Props story="default" />
