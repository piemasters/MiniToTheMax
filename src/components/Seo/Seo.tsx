import React, { FC } from 'react';

export interface SeoProps {
  title: string;
  description: string;
  image: string;
  article?: boolean;
  siteUrl: string;
}

export const Seo: FC<SeoProps> = ({
  title,
  description,
  image,
  article,
  siteUrl,
}) => {
  return (
    <>
      {/* General tags */}
      <meta name="description" content={description} />
      <meta name="image" content={image} />

      {/* OpenGraph tags */}
      {siteUrl && <meta property="og:url" content={siteUrl} />}
      {(article ? true : null) && <meta property="og:type" content="article" />}
      {title && <meta property="og:title" content={title} />}
      {description && <meta property="og:description" content={description} />}
      {image && <meta property="og:image" content={image} />}
    </>
  );
};

export default Seo;
