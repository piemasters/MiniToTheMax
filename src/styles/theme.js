export default {
  colors: {
    text: '#000',
    muted: '#999',
  },
};
