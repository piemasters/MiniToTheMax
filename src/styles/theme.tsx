export interface Theme {
  colors: {
    text: string;
    muted: string;
  };
}

const appTheme: Theme = {
  colors: {
    text: '#000',
    muted: '#999',
  },
};

export default appTheme;
