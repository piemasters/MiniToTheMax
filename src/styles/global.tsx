import { css } from '@emotion/react';
import { Theme } from './theme';
import { resetStyles } from './reset';

export const globalStyles = (theme: Theme) => {
  return [resetStyles, gatsby(theme), blockquote(theme)];
};

const blockquote = (theme: Theme) => css`
  blockquote {
    background: ${theme.colors.lightgrey};
    border-left: 0.7rem solid ${theme.colors.primary};
    border-radius: 5px;
    margin: 1rem;
    padding: 0.8rem;
    quotes: '\\201C''\\201D''\\2018''\\2019';
    &:before {
      color: ${theme.colors.textSecondary};
      content: open-quote;
      font-size: 3rem;
      line-height: 0.1rem;
      margin-right: 1rem;
      vertical-align: -0.4em;
    }
    p {
      color: ${theme.colors.textSecondary};
      display: block;
      font-size: 0.8rem;
      font-style: italic;
      margin-top: 1rem;
      &:first-of-type {
        display: inline;
      }
    }
  }
`;

const gatsby = (theme: Theme) => css`
  .gatsby-resp-image-wrapper {
    width: 100%;
  }
`;
