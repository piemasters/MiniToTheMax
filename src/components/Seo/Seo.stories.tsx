import type { Meta, StoryObj } from '@storybook/react-webpack5';

import { Seo } from './Seo';

const meta: Meta<typeof Seo> = {
  component: Seo,
  argTypes: {
    title: {
      name: 'title',
      description: 'Title of the page',
      defaultValue: 'SEO Title',
      control: { type: 'text' },
    },
    description: {
      name: 'description',
      description: 'A brief summary of the page',
      defaultValue: 'SEO description',
      control: { type: 'text' },
    },
    image: {
      name: 'image',
      description: 'Path to thumbnail used when link is shared',
      defaultValue: 'images/logo.png',
      control: { type: 'text' },
    },
    article: {
      name: 'article',
      description: 'Whether the page type is a blog post',
      defaultValue: false,
      control: { type: 'boolean' },
    },
    siteUrl: {
      name: 'pathname',
      description: 'The unique slug of the page after the base URL',
      defaultValue: '',
      control: { type: 'text' },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Seo>;

export const Primary: Story = {
  args: {
    title: 'SEO Title',
    description: 'Description of the SEO',
    image: 'images/logo.png',
    article: true,
    siteUrl: 'https://minitothemax.app',
  },
};
