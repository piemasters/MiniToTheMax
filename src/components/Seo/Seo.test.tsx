import { render } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';

import { Seo } from './Seo';
import { appTheme } from '../../styles/theme';

describe('Seo', () => {
  test('renders correctly', () => {
    const data = {
      site: {
        siteMetadata: {
          title: 'MiniToTheMax',
          description: 'An Aspiring Hobby & Painting Blog',
          siteUrl: 'https://minitothemax.app',
          image: `content/assets/images/logo.png`,
        },
      },
    };

    const { container } = render(
      <ThemeProvider theme={appTheme}>
        <Seo
          title={data.site.siteMetadata.title}
          description={data.site.siteMetadata.description}
          image={data.site.siteMetadata.image}
          siteUrl={data.site.siteMetadata.siteUrl}
        />
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
