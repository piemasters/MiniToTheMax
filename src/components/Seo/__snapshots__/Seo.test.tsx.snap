// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Seo renders correctly 1`] = `
<div>
  <meta
    content="An Aspiring Hobby & Painting Blog"
    name="description"
  />
  <meta
    content="content/assets/images/logo.webp"
    name="image"
  />
  <meta
    content="https://minitothemax.app"
    property="og:url"
  />
  <meta
    content="MiniToTheMax"
    property="og:title"
  />
  <meta
    content="An Aspiring Hobby & Painting Blog"
    property="og:description"
  />
  <meta
    content="content/assets/images/logo.webp"
    property="og:image"
  />
  <html
    lang="en"
  />
  <title>
    MiniToTheMax
  </title>
</div>
`;
